near/src/config.rs

```diff
@@ -792,13 +651,14 @@ pub fn init_configs(
                 developer_reward_percentage: DEVELOPER_PERCENT,
                 protocol_reward_percentage: PROTOCOL_PERCENT,
                 max_inflation_rate: MAX_INFLATION_RATE,
-                total_supply,
+                total_supply: 0,
```

Comment:
why?

Reply:
I encapsulated the calculation of total_supply into `init` method, but I still need to set some value on initialization.

Reply:
I think `init` then should become `new` and take all the fields as the arguments except `total_supply`.

Reply:
I considered it, but having `new` method with so many arguments is a mess, and I didn't want to go with a builder pattern (common in Rust) just yet...

@nearmax Do you think it is absolutely necessary? Both current and `new` approaches are non-ideal to me. The cleanest solution would be the builder pattern.

Reply:
Implemented Default trait to avoid the confusing `total_supply: 0`